Tasks/MysqlDeploymentOnMachineGroupV1/operations/ToolPathOperations.ts

```diff
@@ -0,0 +1,170 @@
+import task = require('vsts-task-lib/task');
+var winreg = require('winreg');
+import Q = require('q');
+
+export class ToolPathOperations {
+
+    /**
+     * Get installed path of mysql either it is linux or windows 
+     */
+    public async getInstalledPathOfMysql(): Promise<string> {
+        let defer = Q.defer<string>(); 
+        // To check either it is linux or windows platform
+        if(task.osType().match(/^Win/)){
+            this.getInstalledPathOfMysqlForWindow().then((path) => {
+                defer.resolve(path);
+            },(error) =>{
+                task.debug("Error during window mysql path finding: "+ error);
+                defer.reject(task.loc("WindowMysqlClientMissingError"));
+            });
+        }
+        else{
+            // linux check
+            this.getInstalledPathOfMysqlForLinux().then((path) => {
+                defer.resolve(path);
+            },(error) =>{
+                task.debug("Error during linux mysql path finding: "+ error);
+                defer.reject(task.loc("LinuxMysqlClientMissingError"));
+            });  
+        }
+
+        return defer.promise;
+    }
+
+
+    /**
+     * Get installed path of mysql for Linux  
+     */
+    public async getInstalledPathOfMysqlForLinux(): Promise<string> {
+        let defer = Q.defer<string>();
+        try{
+            const path = task.which("mysql", true);
+            defer.resolve(path);
+        }catch(error){
+            defer.reject(error);
+        }
+
+        return defer.promise;
+    }
+
+     /**
+     * Get installed path of mysql for windows 
+     */
+    public async getInstalledPathOfMysqlForWindow(): Promise<string> {
+        let defer = Q.defer<string>();
+        // If user has installed 32 bit mysql client in 64 bit machine
+        this.getInstalledLocationFromPath("\\Software\\Wow6432Node\\MySQL AB").then((path) => {
+            task.debug('Window Wow6432 mysql executable path: '+path);
+            defer.resolve(path + "bin\\mysql.exe");
```

Comment:
use path.join

Reply:
It has escaped this is the reason not using path.join